// react-jsx-highcharts/dist/es/components/Series/Series.js
import _objectSpread from "@babel/runtime/helpers/esm/objectSpread";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _assertThisInitialized from "@babel/runtime/helpers/esm/assertThisInitialized";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import React, { Component } from 'react';
import PropTypes from 'prop-types';
import uuid from 'uuid/v4';
import { isFunction } from 'lodash-es';
import { isEqual } from 'lodash-es';
import { attempt } from 'lodash-es';
import isImmutable from 'is-immutable';
import immutableEqual from 'immutable-is';
import { Provider } from '../SeriesContext';
import addEventProps, { getNonEventHandlerProps } from '../../utils/events';
import getModifiedProps from '../../utils/getModifiedProps';
import { logSeriesErrorMessage } from '../../utils/warnings';

var Series =
/*#__PURE__*/
function (_Component) {
  _inherits(Series, _Component);

  function Series(props) {
    var _this;

    _classCallCheck(this, Series);

    _this = _possibleConstructorReturn(this, _getPrototypeOf(Series).call(this, props));

    _defineProperty(_assertThisInitialized(_assertThisInitialized(_this)), "getSeriesConfig", function () {
      var _this$props = _this.props,
          id = _this$props.id,
          data = _this$props.data,
          requiresAxis = _this$props.requiresAxis,
          getAxis = _this$props.getAxis,
          children = _this$props.children,
          rest = _objectWithoutProperties(_this$props, ["id", "data", "requiresAxis", "getAxis", "children"]);

      var seriesId = isFunction(id) ? id() : id;
      var seriesData = isImmutable(data) ? data.toJS() : data;
      var nonEventProps = getNonEventHandlerProps(rest);

      var config = _objectSpread({
        id: seriesId,
        data: seriesData
      }, nonEventProps);

      if (requiresAxis) {
        var axis = getAxis();
        config[axis.type] = axis.id;
      }

      return config;
    });

    if (process.env.NODE_ENV === 'development') {
      var type = props.type,
          getHighcharts = props.getHighcharts;
      var seriesTypes = Object.keys(getHighcharts().seriesTypes);
      if (seriesTypes.indexOf(type) === -1) logSeriesErrorMessage(type);
    }

    return _this;
  }

  _createClass(Series, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      var chart = this.props.getChart(); // Create Highcharts Series

      var opts = this.getSeriesConfig();
      this.series = chart.addSeries(opts, true);
      var update = this.series.update.bind(this.series);
      addEventProps(update, this.props); // Re-render to pass this.series to Provider

      this.forceUpdate();
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      var _this$props2 = this.props,
          visible = _this$props2.visible,
          data = _this$props2.data,
          rest = _objectWithoutProperties(_this$props2, ["visible", "data"]); // Using setData is more performant than update


      if (isImmutable(data) && immutableEqual(data, prevProps.data) === false) {
        this.series.setData(data.toJS(), true);
      } else if (isEqual(data, prevProps.data) === false) {
        this.series.setData(data, true);
      }

      if (visible !== prevProps.visible) {
        this.series.setVisible(visible);
      }

      var modifiedProps = getModifiedProps(prevProps, rest);

      if (modifiedProps !== false) {
        this.series.update(modifiedProps);
      }
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      if (this.series && this.series.remove) {
        // Series may have already been removed, i.e. when Axis unmounted
        attempt(this.series.remove.bind(this.series));
      }
    }
  }, {
    key: "render",
    value: function render() {
      if (!this.series) return null;
      return React.createElement(Provider, {
        value: this.series
      }, this.props.children);
    }
  }]);

  return Series;
}(Component);

_defineProperty(Series, "defaultProps", {
  type: 'line',
  id: uuid,
  children: null,
  data: [],
  requiresAxis: true,
  visible: true
});

export default Series;
